tests/installation/select_products_sle.pm

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2014-2017 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+# Summary: performing extra actions specific to sle 15 which are not available normally
+# Maintainer: Rodion Iafarov <riafarov@suse.com>
+
+use strict;
+use base "y2logsstep";
+use testapi;
+use utils qw(addon_license sle_version_at_least);
+
+sub run {
+  # SLE 15 has unsigned file errors, workaround them - rbrown 04/07/2017
+    if (sle_version_at_least('15')) {
+        while (check_screen('sle-15-unsigned-file')) {
+            record_soft_failure 'bsc#1047304';
+            send_key 'alt-y';
+        }
+    }
```

Comment:
you copy-pasted that from tests/installation/addon_products_sle.pm, 
[DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself). You are calling the same code multiple times, please [extract method](http://refactoring.com/catalog/extractMethod.html).

Also please don't use check_screen with non-zero value